### PR TITLE
Provide consistent exit codes

### PR DESCRIPTION
**Pull request by bertschneider**
This change uses the possibility to stop the fuzzer loop provided in the patched
libFuzzer version. Incorporating this allows the removal of the stopFuzzing
functions and general simplifications of the fuzzing process.

The core fuzzing function now returns the expected exit code and found error so
that invoking code can handle them according to its needs, e.g. exit the
process with the given code in case of CLI, and rethrow the error in case of
Jest.

### CODE REVIEW

**Review by oetr — COMMENTED** [review]
Thanks for this awesome PR! It makes a large chunk of our native code much clearer.

File: packages/core/finding.ts [hunk 1]
```diff
@@ -42,25 +56,37 @@ export function clearFirstFinding(): Finding | undefined {
...
 	return reportedFinding;
 }
 
+/**
+ * Save the first finding reported by any bug detector.
+ *
+ * @param findingMessage - The finding to be saved.
+ * @param containStack - Whether the finding should contain a stack trace or not.
+ */
+export function reportFinding(
```
[the remark this answers is not in the document]
Comment (bertschneider) [reply packages/core/finding.ts]: Oh, yes, that's way better! Changed!

File: packages/core/core.ts [hunk 2]
```diff
@@ -152,150 +162,166 @@ function getFilteredBugDetectorPaths(
...
 
-function errorName(error: unknown): string {
-	if (error instanceof Error) {
-		// error objects
-		return error.name;
-	} else if (typeof error !== "object") {
-		// primitive types
-		return String(error);
+// Wrap fuzz target to print and dump the crashing input on error.
+function asCrashDumpFuzzFn(fuzzFn: fuzzer.FuzzTarget) {
```
[the remark this answers is not in the document]
Comment (bertschneider) [reply packages/core/core.ts:270]: Exactly, the additional call from `jest-runner` is the problem. I tried to come up with a nice solution but failed. I will give it another try, though, as I also would like to have only one wrapper around the fuzz target.

File: packages/core/core.ts [hunk 3]
```diff
@@ -152,150 +162,166 @@ function getFilteredBugDetectorPaths(
...
-			stopFuzzing(
-				err,
-				options.expectedErrors,
-				options.coverageDirectory,
-				options.coverageReporters,
-				options.sync,
-			);
-		},
-	);
+	const findingAwareFuzzFn = asFindingAwareFuzzFn(fuzzFn);
```
[the remark this answers is not in the document]
Comment (bertschneider) [reply packages/core/core.ts:172]: Actually, it was there before :see_no_evil: My reasoning on extracting the call was that `loadFuzzFunction` did two things, as the name suggests loading the actual fuzz function, but also wrapping that to report findings. I find it a little bit cleaner and easier to follow this way. We could change it back, though, no hard feeling either way.

File: .gitignore [hunk 4]
```diff
@@ -53,4 +54,5 @@ node_modules/
 *.tgz
 
 # Dictionaries generated by Jazzer.js
-.JazzerJs-merged-dictionaries
\ No newline at end of file
+.JazzerJs-merged-dictionaries
```
[the remark this answers is not in the document]
Comment (bertschneider) [reply .gitignore]: Good catch! Deleted.

**Comment by bertschneider** [comment]
I incorporated the latest signal handling changes into this PR. It seems that handling SIGSEGV in the native addon only is sufficient. SIGINT still works without a handler in node, so we can get rid of all JS signal handling code.
Furthermore, I changed the sigsegv error message so that it's recognized by cifuzz. 

This leaves Jazzer.js with consistent error messages, except for timeouts, where libFuzzer kills the process with a timeout crash and exit code `70`, and segfaults, where the addon kills the process with exit code `139`.

**Review by oetr — APPROVED** [review]

File: packages/core/core.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bertschneider) [reply packages/core/core.ts]: Pushed another version without the additional wrapper and enforced types.

**Pull request merged**